pkg/controllers/util/planner/planner.go

```diff
@@ -218,7 +218,7 @@ func getDesiredPlan(
 	// Assign each cluster the minimum number of replicas it requested.
 	for _, preference := range preferences {
 		min := minInt64(preference.MinReplicas, remainingReplicas)
-		if capacity, hasCapacity := estimatedCapacity[preference.clusterName]; hasCapacity && capacity < min {
+		if capacity, hasCapacity := estimatedCapacity[preference.clusterName]; hasCapacity && capacity >= 0 && capacity < min {
```

Comment:
estimatedCapacity is used in a few other places too. Should we also update those?

Reply:
yes, then i think it will be better to do a preprocessing step where we remove all negative estimatedCapacity values